test/integration/ovs/ofctrl_test.go

```diff
@@ -355,6 +355,52 @@ func TestBundleErrorWhenOVSRestart(t *testing.T) {
 	require.Equal(t, 0, failCount, "No fail case expected")
 }
 
+// TestReconnectOFSwitch verifies that the OpenFlow connection to OVS can be restored, even when OVS is down for a long
+// amount of time.
+func TestReconnectOFSwitch(t *testing.T) {
+	br := "br07"
+	err := PrepareOVSBridge(br)
+	require.Nil(t, err, fmt.Sprintf("Failed to prepare OVS bridge: %v", err))
+	defer func() {
+		err = DeleteOVSBridge(br)
+	}()
+
+	bridge := binding.NewOFBridge(br)
+	reconnectCh := make(chan struct{})
+	var connectCount int
+	go func() {
+		for {
+			select {
+			case <-reconnectCh:
+				connectCount++
+			}
+		}
+	}()
+	err = bridge.Connect(maxRetry, reconnectCh)
+	require.Nil(t, err, "Failed to start OFService")
+	defer bridge.Disconnect()
+
+	require.Equal(t, connectCount, 1)
+	// The max delay for the initial connection is 5s. Here we assume the OVS is stopped then started after 8s, and
+	// check that we can re-connect to it after that delay.
+	go func() {
+		DeleteOVSBridge(br)
+		select {
+		case <-time.After(8 * time.Second):
+			err := PrepareOVSBridge(br)
+			require.Nil(t, err, fmt.Sprintf("Failed to prepare OVS bridge: %v", err))
+		}
```

Comment:
This select is normally used to wait for multiple conditions while you have only single condition, it can just be `time.Sleep(8 * time.Second)`.